Tests/RoslynPluginGeneratorTests/AnalyzerPluginGeneratorTests.cs

```diff
@@ -0,0 +1,361 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AnalyzerPluginGeneratorTests.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+using System;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarQube.Plugins.Test.Common;
+using System.IO;
+using NuGet;
+using System.Linq;
+using System.Collections.Generic;
+namespace SonarQube.Plugins.Roslyn.RoslynPluginGeneratorTests
+{
+    /// <summary>
+    /// Tests for NuGetPackageHandler.cs
+    /// </summary>
+    [TestClass]
+    public class AnalyzerPluginGeneratorTests
+    {
+        public TestContext TestContext { get; set; }
+
+        private enum License { Required, NotRequired };
+        private enum Node { A, B, C, D, E, F };
+
+        [TestMethod]
+        public void Generate_PackageNoAccept_NoDependencies_Succeeds()
+        {
+            // Arrange
+            TestLogger logger = new TestLogger();
+            string outputDir = TestUtils.CreateTestDirectory(this.TestContext, ".out");
+            
+            string localNuGetSourceDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGetSource");
+            string localNuGetDownloadDir = TestUtils.CreateTestDirectory(this.TestContext, ".localNuGetDownload");
+            IPackageManager localNuGetSourceStore = CreatePackageManager(localNuGetSourceDir);
+            SetupTestGraph(localNuGetSourceStore);
```

Comment:
The NuGet "source" directory (where packages are created) and the "download" directory (where packages are fetched to by the NuGetPackageHandler) can be moved into simple getter methods, which allows us to genericise the setup a bit more.